docs/content/docs/operations/deployment.md

```diff
@@ -24,18 +24,141 @@ under the License.
 
 ## Local Run with Test Data
 
-{{< hint warning >}}
-**TODO**: How to run with test data with LocalExecutorEnvironment.
-{{< /hint >}}
+After completing the [installation of flink-agents]({{< ref "docs/get-started/installation" >}}) and [building your agent]({{< ref "docs/development/workflow_agent" >}}), you can test and execute your agent locally using a simple Python script. This allows you to validate logic without requiring a Flink cluster.
+
+### Key Features
+
+- **No Flink Required**: Local execution is ideal for development and testing.
+- **Test Data Simulation**: Easily inject mock inputs for validation.
+- **IDE Compatibility**: Run directly in your preferred development environment.
+
+### Data Format
+
+#### Input Data Format
+
+The input data should be a list of dictionaries (`List[Dict[str, Any]]`) with the following structure:
+
+``````python
+[
+    {
+        # Optional field: Context key for multi-session management
+        # Priority order: "key" > "k" > auto-generated UUID
+        "key": "session_001",  # or use shorthand "k": "session_001"
+        
+        # Required field: Input content (supports text, JSON, or any serializable type)
+        # This becomes the `input` field in InputEvent
+        "value": "Calculate the sum of 1 and 2.",  # or shorthand "v": "..."
+    },
+    ...
+]
+``````
+
+#### Output Data Format
+
+The output data is a list of dictionaries (`List[Dict[str, Any]]`) where each dictionary contains a single key-value pair representing the processed result. The structure is generated from `OutputEvent` objects:
+
+``````python
+[
+    {key_1: output_1},  # From first OutputEvent
+    {key_2: output_2},  # From second OutputEvent
+    ...
+]
+``````
+
+Each dictionary in the output list follows this pattern:
+
+``````
+{
+    # Key: Matches the input context key (from "key"/"k" field or auto-generated UUID)
+    # Value: Result from agent processing (type depends on implementation)
+    <context_key>: <processed_output>
+}
+``````
+
+### Example for Local Run with Test Data
```

Comment:
Ok, here comes the example. But normally, we don't look for the examples in the another section. Maybe put the example before the input and output sections? Otherwise, it's super confusing, like we don't even know that are the inputs and outputs for when you start talking about their formats.